src/api/java/com/minecolonies/api/colony/buildings/IBuilding.java

```diff
@@ -32,6 +33,27 @@
 
 public interface IBuilding extends IBuildingContainer, IRequestResolverProvider, IRequester
 {
+    /**
+     * Check if the building has a particular module.
+     * @param clazz the module of the class to check.
+     * @return true if so.
+     */
+    boolean hasModule(final Class<? extends IBuildingModule> clazz);
+
+    /**
+     * Get the module with a particular class.
+     * @param clazz the modules class.
+     * @return the module or null of not existant.
+     */
+    @Nullable
+    <T extends IBuildingModule> T getModule(Class<T> clazz);
```

Comment:
I do not like nullables.
It can cause weird issues.

Use an optional here, which forces people to think about what todo when the module is not available in the expected building automatically.

Reply:
Might additionally collapse the hasModule into this method when switched to the optional.

Computationally it is identical.